File: highlydivisibletriangularnumber.py
import math

def divisors (n):
  divisors = 2;
  for i in range (2, n):
    if (n % i == 0):
      divisors = divisors + 1
  return divisors
  
num = 55
addnum = 10

while (divisors(num) <= 500):
  addnum = addnum + 1
  num = num + addnum


print num      

File: highlydivisibletriangularnumber.cpp
#include <iostream>

using namespace std;

bool divisors_500 (int n)
{
  int divisors = 2;

  for (int i = 2; i < n; i++)
  {
    if (n % i == 0)
    {
      divisors++;
    }
  }
  return (divisors > 500);
}

int main() {

  int num = 55;
  int addnum = 10;

  while (num < 100000)
  {
    addnum++;
    num = num + addnum; 
  }

  while (!divisors_500(num))
  { 
    addnum++;
    num = num + addnum;
  }

  cout << num << endl;      

}